server/src/HttpServer.hs

```diff
@@ -15,26 +16,35 @@ import qualified Data.Text.Lazy as LText
 import qualified Network.Wai as Wai
 import qualified Web.Scotty.Trans as Scotty
 
+import qualified System.Log.Raven.Types as Sentry
+
 import JwtMiddleware (jwtMiddleware)
 import Core (Core (..), EnqueueResult (..))
 import Config (Config (..))
 import qualified Store
 import qualified Core
 import qualified Metrics
+import qualified SentryLogging
 
-new :: Core -> IO Application
-new core =
+new :: Maybe Sentry.SentryService -> Core -> IO Application
+new mSentryService core =
   scottyApp $ do
     -- first middleware is the outermost. this has to be the metrics middleware
     -- in order to intercept all requests their corresponding responses
     forM_ (coreMetrics core) $ middleware . metricsMiddleware
-
+    -- Use the Sentry logger if available
+    -- Scottys error handler will only catch errors that are thrown from within
+    -- a ```liftAndCatchIO``` function.
+    if isJust mSentryService then
+        Scotty.defaultHandler (liftIO . SentryLogging.logCrashMessage "Thread handler error" mSentryService . show)
+    else
+        Scotty.defaultHandler (liftIO . putStrLn . LText.unpack )
```

Comment:
Instead of printing directly to stdout, it should go via the global logger (which you can get via `coreLogger core`).
That way, we can control what to do with log messages in one place.